automake.sh

```diff
@@ -8,54 +9,73 @@ else
   make -f Makefile.ref DEFINES=-DHAVE_VA_LIST_AS_ARRAY CC=gcc
 fi
 
-$ev=$?
+ev=$?
 if [ $ev -ne 0 ]; then
-  echo "Unable to build spidermonkey, cannot continue"
+  echo "Unable to build SpiderMonkey! Exiting..."
   exit $ev
+else
+  echo "Done building SpiderMonkey."
 fi
 
-if [ "$(uname)" = "Darwin" ]
+echo "Creating JavaScript library..."
+if [ "$(uname)" = "Darwin" ] # macOS
 then
-        # Mac OS X
-        ar rcs libjs32.a Darwin_DBG.OBJ/*.o
-        cp Darwin_DBG.OBJ/jsautocfg.h ./
+  libtool -static -o libjs32.a -s Darwin_DBG.OBJ/*.o
+  cp Darwin_DBG.OBJ/jsautocfg.h ./
 elif [ "$(uname)" = "FreeBSD" ]
 then
-        ar rcs libjs32.a FreeBSD_DBG.OBJ/*.o
-        cp FreeBSD_DBG.OBJ/jsautocfg.h ./
+  ar rcs libjs32.a FreeBSD_DBG.OBJ/*.o
+  cp FreeBSD_DBG.OBJ/jsautocfg.h ./
 elif [ "$(expr substr $(uname -s) 1 5)" = "Linux" ]
 then
-        # Linux
-        ar -r libjs32.a Linux_All_DBG.OBJ/*.o
-        cp Linux_All_DBG.OBJ/jsautocfg.h ./
+  # Linux
+  ar -r libjs32.a Linux_All_DBG.OBJ/*.o
+  cp Linux_All_DBG.OBJ/jsautocfg.h ./
+fi
+
+ev=$?
+if [ $ev -ne 0 ]; then
+  echo "Unable to create JavaScript library! Exiting..."
+  exit $ev
+else
+  echo "Done creating JavaScript library."
 fi
 
+echo "Bulding zlib..."
 cd ../zlib
-echo "Bulding zlib"
 make distclean
-./configure
-make
+./configure && make
 
-$ev=$?
+ev=$?
 if [ $ev -ne 0 ]; then
-  echo "Unable to build zlib, cannot continue"
+  echo "Unable to build zlib! Exiting..."
   exit $ev
+else
+  echo "Done building zlib."
 fi
 
+echo "Building UOX3..."
 cd ../source
-echo "Building UOX3"
 if [ "$(uname)" = "FreeBSD" ]
 then
   gmake
 else
   make
 fi
+
 ev=$?
-if [ -f ./uox3 ]; then
-  cp uox3 ..
-  echo "Done! You should now find the compiled uox3 binary in the root UOX3 project directory. Copy this binary to a separate directory dedicated to running your UOX3 shard, along with the contents of the UOX3/data directory, to avoid potential git conflicts and accidental overwriting of data when pulling UOX3 updates in the future."
+if [ $ev -ne 0 ]; then
+  echo "Unable to build UOX3! Exiting..."
+  exit $ev
 else
-  echo "uox3 output not created!  Please review compile status"
+  if [ -f ./uox3 ]; then
+    cp uox3 ..
+    echo "Done! You should now find the compiled uox3 binary in the root UOX3 project directory. Copy this binary"
+    echo "to a separate directory dedicated to running your UOX3 shard, along with the contents of the UOX3/data directory,"
+    echo "to avoid potential git conflicts and accidental overwriting of data when pulling UOX3 updates in the future."
+  else
```

Comment:
How do you envisage this else ever triggering?  How could make pass, but not result in a valid output?

Reply:
Are you suggesting removal of the file check prior to copying too? I say it is better safe than sorry and to not make any assumptions. It's like having a default safety net to handle the possibility. Perhaps some nameless security tool flags `uox3` as malicious (because the program is unknown and listens for inbound connections on a socket) and removes the binary immedlately following its generation. If the file can exist, it also seems logical that the opposite case could conceivably occur. Lastly, it is merely two additional lines in a shell script -- I do not feel there is any harm in keeping the `else` and `echo` lines intact.

Reply:
Resolving this for now, can revisit later if needed.